rfcs/20190729-keras-preprocessing.md

```diff
@@ -0,0 +1,732 @@
+            Default is 'nearest'.
+            Points outside the boundaries of the input are filled
+            according to the given mode:
+            - 'constant': kkkkkkkk|abcd|kkkkkkkk (cval=k)
+            - 'nearest':  aaaaaaaa|abcd|dddddddd
+            - 'reflect':  abcddcba|abcd|dcbaabcd
+            - 'wrap':  abcdabcd|abcd|abcdabcd
+        fill_value: Float or Int.
+            Value used for points outside the boundaries
+            when `fill_mode = "constant"`.
+        horizontal_flip: Boolean. Randomly flip inputs horizontally.
+        vertical_flip: Boolean. Randomly flip inputs vertically.
+        rescale: rescaling factor. Defaults to None.
+            If None or 0, no rescaling is applied,
+            otherwise we multiply the data by the value provided
+            (after applying all other transformations).
+        resize: Tuple of int (height, width). Dimensions to resize
+            images to.
+        preprocessing_function: function that will be applied on each input.
+            The function will run after the image is resized and augmented.
+            The function should take one argument:
+            one image (Numpy tensor with rank 3),
+            and should output a Numpy tensor with the same shape.
+        data_format: Image data format,
+            either "channels_first" or "channels_last".
+            "channels_last" mode means that the images should have shape
+            `(samples, height, width, channels)`,
+            "channels_first" mode means that the images should have shape
+            `(samples, channels, height, width)`.
+            It defaults to the `image_data_format` value found in your
+            Keras config file at `~/.keras/keras.json`.
+            If you never set it, then it will be "channels_last".
+        interpolation_order: int, order to use for
+            the spline interpolation. Higher is slower.
+        dtype: Dtype to use for the generated arrays.
+    """
+```
+
+Notes:
+
+1) The meaning of existing arguments does not change.
+
+2) The `*_range` arguments are polymorphic (no change from current state):
+- Float value should be between 0 and 1 and indicates a fraction (e.g. 0.2 indicates "-20% +20%" variation).
+- Integer value N indicates a [-N +N] variation interval, where N is either pixels or degrees depending on use case.
+- Tuple of 2 floats or 2 integers indicates 2 different boundaries in the intervals above.
+
+3) Data augmentation happens on actual images, before normalization (this is more correct, and necessary in order to visualize the effect of augmentation).
+
+
+#### New additions
+
+```python
+@classmethod
+def from_config(self, config):
+    # Returns instance
+
+def get_config(self, config):
+    # Returns JSON-serializable config dict
+
+def preview(self, data, save_to_directory=None, save_prefix=None, save_format='png'):
+    """Enables users to previous the image augmentation configuration.
```

Comment:
Enables users to **preview** the image augmentation configuration.

Reply:
Fixed